orb-ui/src/engine/mod.rs

```diff
@@ -364,11 +364,45 @@ impl AnimationState {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq)]
+#[allow(dead_code)]
+pub enum Transition {
+    /// Starting transitions
+
+    /// Fade in the animation with a duration.
+    FadeIn(f64),
+    /// Launch the animation after a delay.
+    StartDelay(f64),
+    /// Shrink animated segments to zero or target size
+    Shrink,
+
+    /// Stopping transitions
+
+    /// immediately stop the animation
+    ForceStop,
+    /// fade out the animation with a duration.
+    FadeOut(f64),
+    /// play the animation one last time
+    PlayOnce,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum TransitionStatus {
+    /// The transition exists and will be nicely handled.
+    Smooth,
+    /// The new animation will abruptly replace the current one.
+    Sharp,
+}
+
 /// Generic animation.
 pub trait Animation: Send + 'static {
     /// Animation frame type.
     type Frame;
 
+    fn name(&self) -> &str {
```

Comment:
consider reworking to be kind(&self) -> AnimationKind, where AnimationKind is an enum with one variant for each animation type.

Reply:
decided to do in another PR.